### PR TITLE
Nit fixes

### PR DESCRIPTION
**Pull request by jaadams5**
1) I noticed I was getting "Unknown skill ID in charsheet.php: 17".   Perhaps, since KoL is reporting it, the comment in the file is incorrect?

2) autumnaton locations was generating NPEs.  I fixed them by adding locations.

Both of these were discovered with a decades old, never-ascended character so they certainly are edge cases.  There may be a better way to deal with the locations.  I just wanted things to work so a script would run for both of my characters.  If this fix remains then I can certainly space the location and environment data for the OBE adventure locations.

### CODE REVIEW

File: src/data/adventures.txt [hunk 1]
```diff
@@ -705,3 +705,23 @@ PirateRealm Island	adventure=531	DiffLevel: mid Env: outdoor Stat: 0 nowander	St
 PirateRealm Island	adventure=531	DiffLevel: mid Env: outdoor Stat: 0 nowander	Red Roger's Fortress
 PirateRealm Island	adventure=531	DiffLevel: mid Env: outdoor Stat: 0 nowander	Jack's Hideout
 PirateRealm Island	adventure=531	DiffLevel: mid Env: outdoor Stat: 0 nowander	The Temple
+
+# Obselete areas that nevertheless are reported as possibilities for the autumnaton
```
Comment (midgleyc) [inline src/data/adventures.txt:709]:
```suggestion
# Obsolete areas that nevertheless are reported as possibilities for the autumnaton
```

File: src/data/adventures.txt [hunk 2]
```diff
@@ -705,3 +705,23 @@ PirateRealm Island	adventure=531	DiffLevel: mid Env: outdoor Stat: 0 nowander	St
...
+Plains	adventure=18	DiffLevel: unknown Env: unknown Stat: 10	Degrassi Knoll OBE
```
Comment (midgleyc) [inline src/data/adventures.txt:710]: Can these be consistently `(OLD)` (or consistently one thing)? Is OBE obsolete?

**Review by midgleyc — CHANGES_REQUESTED** [review]
It would be nice to fix the NPE on autumnaton, too. Is it in `adv.getEnvironment()`, because `adv` can be null?

**Comment by midgleyc** [comment]
Can you post some HTML for using the autumn-aton with the character where Mafia crashes? Would be nice for testing.

File: src/data/adventures.txt (hunk 2 above)
Comment (Veracity0) [inline src/data/adventures.txt:710]: OBE? "Order of the British Empire"? "Out of Body Experience"?

Make them all (OLD)

What happens if you go to adventure.php?snarfblat=18?
I assume it fails.

I feel strongly that unless you can actually adventure there, they do not belong in "Plains" or "Manor2" or whatever.

canAdventure purports to handle EVERY adventure that is in a particular zone.
These do not belong in those switch statements.

Make a new zone in zonelists.txt - Obsolete - with parent Removed.
Make all these locations have zone Obsolete

File: src/data/classskills.txt [hunk 3]
```diff
@@ -34,7 +34,7 @@
 # Skill names have changed, keeping this for historical reasons only
 
 # 16	Summon Snowcone		1	5	0
-# 17	Summon Hilarious Objects		1	5	0
+17	Summon Hilarious Objects		1	5	0
```
Comment (Veracity0) [inline src/data/classskills.txt:37]: Absolutely not.

```
7226	Summon Hilarious Objects	book2.gif	1	5	0
```

What do you get if you look up "Summon Hilarious Objects" by name.

Keep this as a comment.

**Review by Veracity0 — CHANGES_REQUESTED** [review]

File: src/data/adventures.txt (hunk 1 above)
Comment (Veracity0) [reply src/data/adventures.txt:709]: As long as you are doing that, move this line from farther up in the file
```
# Woods	adventure=99	DiffLevel: unknown Env: outdoor	The Road to White Citadel
```
and do the same thing to it.

**Review by Veracity0 — CHANGES_REQUESTED** [review]

File: src/data/adventures.txt (hunk 2 above)
Comment (jaadams5) [reply src/data/adventures.txt:710]: OBE is overtaken by events which apparently doesn't mean that to people who have not worked around the US military.  I originally did not have OBE in the names and about 10 tests failed.  I guessed the tests expected names to be unique and so added something that would identify "my" additions if not.  I'll make consistent.

File: src/data/classskills.txt (hunk 3 above)
Comment (jaadams5) [reply src/data/classskills.txt:37]: Multiple matches.

Comment (jaadams5) [reply src/data/classskills.txt]: Reverted.  Will direct my sense of something needs to be fixed by adding to the CharSheet testing.

**Comment by jaadams5** [comment]
> It would be nice to fix the NPE on autumnaton, too. Is it in `adv.getEnvironment()`, because `adv` can be null?

Have not seen that NPE but I can play with what gets returned as locations and see if I can make it happen.

File: src/data/adventures.txt (hunk 2 above)
Comment (jaadams5) [reply src/data/adventures.txt]: OK on Zones.  Will work on later.

**Comment by jaadams5** [comment]
OK.  I will grab some html and add to the CharPane parsing test so that at least there is documentation that KoL still includes and obsolete skill id.

I will see if I can send an autumnaton to any of the obsolete areas.

I will see if I can generate an NPE accessing environment but I think this fixes that.

We should be prepared for more obsolete adventures and the possibility that an autumnaton can be sent somewhere that cannot be adventured in any more.

**Comment by jaadams5** [comment]
> Can you post some HTML for using the autumn-aton with the character where Mafia crashes? Would be nice for testing.

This is what I grabbed when get_autumnaton_locations() was throwing NPEs

[Response.txt](https://github.com/kolmafia/kolmafia/files/9883191/Response.txt)

**Comment by midgleyc** [comment]
Oh interesting, so by the code, `OLD X` and `X (OLD)` are actually the KoL names.

**Comment by jaadams5** [comment]
> Oh interesting, so by the code, `OLD X` and `X (OLD)` are actually the KoL names.

Yeah, I guess they are.  I can undo the "consistency"  changes if we want.

**Comment by jaadams5** [comment]
OK.  I think I have addressed requested changes.

1) Given that some of the obsolete place names are what is used by KoL should I undo the changes made for consistency and use the KoL name if present?

2) I manually sent the autumnaton to several of the obsolete places and things worked as expected.  I believe KoL will allow the autumnaton to be sent to places where the character cannot adventure.

3) I have HTML from the charpane so I can write a failing test for KoL using a skill we don't expect.  I displayed the page and clicked on Summon Hilarious Objects to get a description and got skill not found.  Now that write that I would like to try it in vanilla KoL but it does seem as if there is a KoL bug.

4) I also have HTML from the autumnatom "sending" page but, besides gleaning locations, I'm not quite sure what I do with it.

**Comment by midgleyc** [comment]
I plan to use the HTML, away from this branch, to test and fix the NPE.

I think it would be nice to use the KoL name for locations, at least if it doesn't conflict with another location. For items, we required prepending with the ID: it might even be nice to do that for locations, too, now that we can know the actual names. This would affect some existing locations like the Hippy Camp.

Here we have "The Arid, Extra-Dry Desert" three times, twice obsolete. I don't know what should be done here.

**Comment by Veracity0** [comment]
We have 2 items named "Staff of Ed". You can disambiguate by:

[2325]Staff of Ed
[7961]Staff of Ed
```
> ashq print($item[staff of ed].to_int())

Multiple matches for "staff of ed"; using "[7961]Staff of Ed". (char 13 to char 24) Clarify by using one of:
$item[[2325]Staff of Ed]
$item[[7961]Staff of Ed]
7961
```
It gives you warnings, but you get the highest item ID.

I suppose we could do that with locations, too. I imagine the highest snarfblat is the current one.

We should use actual KoL names, if possible.

**Comment by midgleyc** [comment]
I think I would prefer the following set of names:
```
Obsolete	adventure=18	DiffLevel: unknown Env: unknown Stat: 10	Degrassi Knoll
Obsolete	adventure=99	DiffLevel: unknown Env: unknown	Stat: 20	The Road to White Citadel
Obsolete	adventure=102	DiffLevel: unknown Env: unknown Stat: 20	The Haunted Kitchen (OLD)
Obsolete	adventure=103	DiffLevel: unknown Env: unknown Stat: 20	The Haunted Conservatory (OLD)
Obsolete	adventure=105	DiffLevel: unknown Env: unknown Stat: 20	The Haunted Billiards Room (OLD)
Obsolete	adventure=107	DiffLevel: unknown Env: unknown Stat: 40	The Haunted Bathroom (OLD)
Obsolete	adventure=108	DiffLevel: unknown Env: unknown Stat: 40	The Haunted Bedroom (OLD)
Obsolete	adventure=111	DiffLevel: unknown Env: unknown Stat: 110	The Black Forest (OLD)
Obsolete	adventure=118	DiffLevel: unknown Env: unknown	Stat: 120	The Hidden City
Obsolete	adventure=119	DiffLevel: unknown Env: unknown Stat: 120	The Palindome (OLD)
Obsolete	adventure=121	DiffLevel: unknown Env: unknown Stat: 120	The Arid, Extra-Dry Desert (Dehydrated)
Obsolete	adventure=123	DiffLevel: unknown Env: unknown Stat: 120	The Arid, Extra-Dry Desert (Ultrahydrated)
Obsolete	adventure=124	DiffLevel: unknown Env: unknown Stat: 150	OLD The Upper Chamber
Obsolete	adventure=125	DiffLevel: unknown Env: unknown Stat: 140	OLD The Middle Chamber
Obsolete	adventure=178	DiffLevel: unknown Env: unknown Stat: 104	The Haunted Wine Cellar (Northwest)
Obsolete	adventure=179	DiffLevel: unknown Env: unknown Stat: 104	The Haunted Wine Cellar (Northeast)
Obsolete	adventure=180	DiffLevel: unknown Env: unknown Stat: 104	The Haunted Wine Cellar (Southwest)
Obsolete	adventure=181	DiffLevel: unknown Env: unknown Stat: 104	The Haunted Wine Cellar (Southeast)
Obsolete	adventure=213	DiffLevel: unknown Env: unknown Stat: 0	Nemesis Cave
```

That is, use the KoL names where it doesn't conflict with an existing location. Where it does (only The Arid, Extra-Dry Desert), use the wiki names.

**Comment by Veracity0** [comment]
Fun fact:
```
Video Game Dungeon	adventure=319	DiffLevel: mid Env: indoor Stat: 0	Video Game Level 1
Video Game Dungeon	adventure=320	DiffLevel: unknown Env: indoor Stat: 0	Video Game Level 2
Video Game Dungeon	adventure=321	DiffLevel: unknown Env: indoor Stat: 0	Video Game Level 3
```
I didn't send it to Level 1, since we already had the DiffLevel, but end of battle messages:
```
Only 22 more adventures before it returns from [DungeonFAQ - Level 2].
It should return from [DungeonFAQ - Level 3] after 33 more adventures.
```
More adventures where our name is not the same as KoL's internal name.

**Review by midgleyc — DISMISSED** [review]

**Review by midgleyc — APPROVED** [review]

**Pull request merged**